analyzers/src/SonarAnalyzer.CSharp/Rules/ImplementSerializationMethodsCorrectly.cs

```diff
@@ -48,5 +48,12 @@ public sealed class ImplementSerializationMethodsCorrectly : ImplementSerializat
                 .FirstOrDefault()
                 ?.Identifier
                 .GetLocation();
+
+        protected override bool ContainsMatchingAttributes(IMethodSymbol methodSymbol) =>
+            base.ContainsMatchingAttributes(methodSymbol) && !HiddenByEditorBrowsableAttribute(methodSymbol);
+
+        private static bool HiddenByEditorBrowsableAttribute(IMethodSymbol methodSymbol) =>
+            methodSymbol.GetAttributes(KnownType.System_ComponentModel_EditorBrowsableAttribute)
+                .Any(x => x.ConstructorArguments.Any(a => 1.CompareTo(a.Value) == 0));
```

Comment:
Can't we check the editor attributes in the base class to have this issue fixed also for VB?

Reply:
You are right, initially I was not able to find that it is also applicable to VB, that's why I've moved the implementation to CS part. Will fix it!